### PR TITLE
CloudStack documentation

### PR DESCRIPTION
**Pull request by sebgoa**
Some brand new docs for cloudstack

### CODE REVIEW

File: docs/compute/drivers/cloudstack.rst [hunk 1]
```diff
@@ -0,0 +1,170 @@
+CloudStack Driver Documentation
+===============================
+
+`CloudStack`_ is an Apache Software Foundation open source software designed to deploy and manage large networks of virtual machines, as a highly available, highly scalable Infrastructure as a Service (IaaS) cloud computing platform. CloudStack is used by a number of service providers to offer public cloud services, and by many companies to provide an on-premises (private) cloud offering, or as part of a hybrid cloud solution.
```
Comment (Kami) [inline docs/compute/drivers/cloudstack.rst]: Please wrap lines on 80 characters.

File: docs/compute/drivers/cloudstack.rst [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+* ``key`` - Your CloudStack API key
+* ``secret`` - Your CloudStack secret key
+* ``host`` - The host of your CloudStack endpoint (e.g `localhost` for http://localhost:8080/client/api) 
+* ``path`` - The path to your CloudStack endpoint (e.g `/client/api` for http://localhost:8080/client/api)
+* ``url`` - The url to your CloudStack endpoint, mutually exclusive with `host` and `path`
+* ``secure`` - True or False. True by default
+
+Typically this will lead to:
+
+::
```
Comment (Kami) [inline docs/compute/drivers/cloudstack.rst]: For consistency, please use the following format for embedded code snippets:

``` pyton
.. sourcecode:: python

    host = 'example.com'
    path = '/path/to/api'
    Driver = get_driver(Provider.CLOUDSTACK)
    conn=Driver(key=apikey,secret=secretkey,host=host,path=path)
```

File: docs/compute/drivers/cloudstack.rst [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+   	host = 'example.com'
+	path = '/path/to/api'
+	Driver = get_driver(Provider.CLOUDSTACK)
+	conn=Driver(key=apikey,secret=secretkey,host=host,path=path)
```
Comment (Kami) [inline docs/compute/drivers/cloudstack.rst]: Please use spaces around assignment operator and after comma (`conn = Driver(key=apikey, secret=secretkey, host=host, path=path)`).

Comment (Kami) [inline docs/compute/drivers/cloudstack.rst]: Even better, move all of the inline snippets to a file and use `literalinclude` so those file can be linted as part of the test run.

**Comment by Kami** [comment]
The tests are failing - https://travis-ci.org/apache/libcloud/jobs/15345517, examples need some lint love :)

File: docs/compute/drivers/cloudstack.rst (hunk 3 above)
Comment (sebgoa) [inline docs/compute/drivers/cloudstack.rst]: and stick them in the examples dir ? or the cloudstack doc dir ?

Comment (Kami) [inline docs/compute/drivers/cloudstack.rst]: Cloudstack examples dir - `docs/examples/compute/cloudstack/`

**Comment by Kami** [comment]
@runseb Merged into trunk. Thanks!

**Comment by sebgoa** [comment]
cool, thanks

**Pull request closed without merging**